keps/sig-node/20190322-image-gc-whitelist.md

```diff
@@ -0,0 +1,273 @@
+---
+title: Image garbage collection whitelist
+authors:
+  - "@libesz"
+  - "@CsatariGergely"
+  - "@BenTheElder"
+owning-sig: sig-node
+participating-sigs:
+- sig-node
+- sig-testing
+reviewers:
+  - "@zhangmingld"
+  - "@Monkeyanator"
+  - "@sjenning"
+  - "@Random-Liu"
+  - "@timothysc"
+approvers:
+  - sig-node-leads
+editor: "@BenTheElder"
+creation-date: 2019-04-26
+last-updated: 2020-04-27
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# image-garbage-collector-whitelist
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [Details of Story 1](#details-of-story-1)
+    - [Details of Story 2](#details-of-story-2)
+  - [Risks and Mitigations](#risks-and-mitigations)
+    - [Risks of Story 1](#risks-of-story-1)
+    - [Risks of Story 2](#risks-of-story-2)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Examples](#examples)
+      - [Alpha -&gt; Beta Graduation](#alpha---beta-graduation)
+      - [Beta -&gt; GA Graduation](#beta---ga-graduation)
+      - [Removing a deprecated flag](#removing-a-deprecated-flag)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks [optional]](#drawbacks-optional)
+- [Alternatives [optional]](#alternatives-optional)
+- [Infrastructure Needed [optional]](#infrastructure-needed-optional)
+<!-- /toc -->
+
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+Enhancing Kubelet image garbage collector with a whitelist of container images. 
+Whitelisted images will be ignored during garbage collection.
```

Comment:
This describes the opposite of the canonical meaning of whitelist (things in a whitelist are the only ones acted on) - perhaps rename this to "keep list"?

Reply:
yeah, I was thinking about this as well, the name seems backwards, this is more of a image GC "blacklist"